uwblueprint/paramedics-web
# Added role validators to resolvers

[Notion ticket](https://www.notion.so/uwblueprintexecs/Backend-Authorization-0f29dae7aad449e98ab43ee1e7f28316)

## Changes


- Added role validators to:
Ambulances,
Collection Points,
Events,
Hospitals,
Location Pins,
Patients,
Users,



- Added the implementation of the role validator to `validator.js`
- Added demoRole for testing
-  Added enumerations for roles:
COMMANDER,
SUPERVISOR,
DISPATCH,




## Screenshots

## Testing
- Test by changing the role in`demoRole` in `validator.js`
- Check each resolver through GraphQL to see if it catches roles that do not have sufficient permissions
- Check that each resolver worked if the demo role had sufficient permission 
- Made sure the role argument for each resolver matches the spreadsheet permissions

While testing, please consult [this](https://docs.google.com/spreadsheets/d/1JTQlflqYCW8p78ceiyYTEJxTe1UuTgLT7OhHtPeyyV8/edit#gid=1413489646) spreadsheet regarding the permissions associated with each role.


## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
